### DISCUSSION TITLE
Swift 5.7 improvements

### DISCUSSION THREAD

**Discussion by randomeizer** [pointfreeco/swift-parsing#218] [General]
I see you have been busy on improving the result builders for 5.7, looks good. Are you thinking about using the new `some`/`any` syntax, or is it not particularly helpful in this context?

I am interested in seeing if some of the more excessive parser signatures that I had to wrap in `AnyParser` just to get it compiling will work better now, since it seems like there have been some under-the-hood performance improvements there too.

**Comment by stephencelis** [comment]
@randomeizer Ha just to make sure I understand, what are you referring to us being busy on? 😅

We definitely have a bunch of ideas for 5.7, which includes `buildPartialBlock`, exploring `some` syntax, as well as `Regex` and `ParseableFormatStyle` integrations.

Anything else come to mind for us to explore?

**Reply by randomeizer** [reply]
This branch, specifically: https://github.com/pointfreeco/swift-parsing/tree/swift-5-7, the `buildPartialBlock` code was most obvious. I will experiment with it a bit soon hopefully - I had some thoughts a while back on a "never fails" parser style, but it was going to require a bunch of generated overloads and didn't pursue it at the time.

Yeah, having some integration with Regex also occurred to me. Looking forward to the performance comparisons also ;)
